### PR TITLE
fix: optional keys and address in eventfilters

### PR DESCRIPTION
**Pull request by PhilippeR26**
## Motivation and Resolution
Solve issue #692 

### RPC version (if applicable)

In EVENT_FILTER, keys and address should be optional.

## Usage related changes

Keys and Address filters are not mandatory.

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

**Review by penovicp — APPROVED** [review]

**Comment by tabaktoni** [comment]
Originally we had as a specification sad none was required, I don't know why v4 set ti to required, but none of it should be required

File: __tests__/rpcProvider.test.ts [hunk 1]
```diff
@@ -151,8 +151,17 @@ describeIfRpc('RPCProvider', () => {
           chunk_size: 2,
         });
 
+        const result1 = await rpcProvider.getEvents({
+          from_block: { block_number: 0 },
+          to_block: { block_number: blockNumber },
+          // address: deploy.contract_address,
```
Comment (tabaktoni) [inline __tests__/rpcProvider.test.ts:157]: You can leave optional parameters for testing

File: src/types/api/openrpc.ts [hunk 2]
```diff
@@ -309,8 +309,8 @@ type STORAGE_KEY = string;
 type EVENT_FILTER = {
   from_block: BLOCK_ID;
   to_block: BLOCK_ID;
-  address: ADDRESS;
-  keys: Array<Array<FELT>>;
+  address?: ADDRESS;
+  keys?: Array<Array<FELT>>;
```
Comment (tabaktoni) [inline src/types/api/openrpc.ts:313]: All parameters should be optional by specification.

**Review by tabaktoni — COMMENTED** [review]
I will do implementation on 0.4 so this will be changed based on specification

**Comment by PhilippeR26** [comment]
Could you finalize the PR? (I am in the train to Paris)

**Comment by tabaktoni** [comment]
> Could you finalize the PR? (I am in the train to Paris)

I will do RPC 0.4 PR and deal with it, don't worry

**Review by tabaktoni — APPROVED** [review]

**Pull request merged**